core/db/mongo/schedule_event.go

```diff
@@ -72,10 +73,14 @@ func (mse *MongoScheduleEvent) SetBSON(raw bson.Raw) error {
 	mse.Service = decoded.Service
 
 	// De-reference the DBRef fields
-	ds := DS.dataStore()
-	defer ds.s.Close()
+	m, err := getCurrentMongoClient()
+	if err != nil {
+		return err
+	}
+	s := m.session.Copy()
+	defer s.Close()
```

Comment:
In event.go isn't used m.session.Copy() and the other related stuff.
Which is the best way to get the values? Maybe the criteria should be unified in another PR